test/base/map.js

```diff
@@ -806,3 +806,20 @@ test("#2253 Map with replace method with intercept", () => {
     expect(observedMap._keys.slice()).toEqual(["prop1", "prop2"])
     expect(observedMap.toJSON()).toEqual({ prop1: 1, prop2: 3 })
 })
+
+test("#2253 filled Map replace method with intercept", () => {
+    const data = {
+        prop4: 4,
+        propIgnored: "ignored",
+        prop5: 5,
+        prop6: 6
+    }
+    const observedMap = map({ prop4: 1, prop2: 2, prop3: 3 })
```

Comment:
@FredyC Like this ? not sure to really understand this [#2253 (comment)](https://github.com/mobxjs/mobx/issues/2253#issuecomment-572977071)

Reply:
@urugator What do you think? Is it sufficient or did I misunderstood something as well?